salesforce/salesforce-inbound-endpoint/salesforce-inbound-endpoint-1.0.0/src/main/java/org/wso2/carbon/inbound/salesforce/poll/SalesforceStreamData.java

```diff
@@ -0,0 +1,299 @@
+        try {
+            client = makeClient();
+        } catch (Exception e) {
+            handleException(e.getMessage());
+        }
+        client.getChannel(Channel.META_HANDSHAKE).addListener(new ClientSessionChannel.MessageListener() {
+            public void onMessage(ClientSessionChannel channel, Message message) {
+                log.info("[CHANNEL:META_HANDSHAKE]: " + message);
+                boolean success = message.isSuccessful();
+                if (!success) {
+                    String error = (String) message.get("error");
+                    if (error != null) {
+                        handleException("Error during HANDSHAKE: " + error);
+                    }
+                    Exception exception = (Exception) message.get("exception");
+                    if (exception != null) {
+                        handleException("Exception during HANDSHAKE: ", exception);
+                    }
+                }
+            }
+        });
+
+        client.getChannel(Channel.META_CONNECT).addListener(new ClientSessionChannel.MessageListener() {
+            public void onMessage(ClientSessionChannel channel, Message message) {
+                log.info("[CHANNEL:META_CONNECT]: " + message);
+                boolean success = message.isSuccessful();
+                if (!success) {
+                    client.disconnect();
+                    log.info("Waiting to Connect with Salesforce Streaming API......");
+                    makeConnect();
+                    String error = (String) message.get("error");
+                    if (error != null) {
+                        handleException("Error during CONNECT: " + error);
+                    }
+                }
+            }
+        });
+
+        client.getChannel(Channel.META_SUBSCRIBE).addListener(new ClientSessionChannel.MessageListener() {
+            public void onMessage(ClientSessionChannel channel, Message message) {
+                log.info("[CHANNEL:META_SUBSCRIBE]: " + message);
+                boolean success = message.isSuccessful();
+                if (!success) {
+                    String error = (String) message.get("error");
+                    if (error != null) {
+                        handleException("Error during SUBSCRIBE: " + error);
+                    }
+                }
+            }
+        });
+
+        client.handshake();
+        boolean handshaken = client.waitFor(waitTime, BayeuxClient.State.CONNECTED);
+        if (!handshaken) {
+            handleException("Failed to handshake: " + client);
+        }
+
+        client.getChannel(channel).subscribe(new MessageListener() {
+            public void onMessage(ClientSessionChannel channel, Message message) {
+                injectSalesforceMessage(message);
+            }
+        });
+    }
+
+    /**
+     * Create a http client.
+     *
+     * @return
+     * @throws Exception
+     */
+    private BayeuxClient makeClient() throws Exception {
```

Comment:
Yes make scense. Also we are catching some exceptions and throwing some other exceptions. We only need to do one at a time IMO